packages/shared/src/components/profile/TopReaderWidget.tsx

```diff
@@ -0,0 +1,102 @@
+import { useQuery } from '@tanstack/react-query';
+import React, { ReactElement } from 'react';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { fetchTopReaders } from '../../lib/topReader';
+import { disabledRefetch } from '../../lib/func';
+import { ActivityContainer, ActivitySectionHeader } from './ActivitySection';
+import { topReaderBadgeDocs } from '../../lib/constants';
+import { MedalBadgeIcon } from '../icons';
+import { IconSize } from '../Icon';
+import { BadgeIconGoldGradient } from '../badges/BadgeIcon';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../typography/Typography';
+import { formatDate, TimeFormatType } from '../../lib/dateFormat';
+import type { PublicProfile } from '../../lib/user';
+
+export const TopReaderWidget = ({
+  user,
+}: {
+  user: PublicProfile;
+}): ReactElement => {
+  const { data: topReader, isLoading: isTopReaderLoading } = useQuery({
+    queryKey: generateQueryKey(RequestKey.TopReaderBadge, user, 'latest'),
+    queryFn: async () => {
+      return await fetchTopReaders(5, user.id);
+    },
+    staleTime: StaleTime.OneHour,
+    ...disabledRefetch,
+  });
+
+  if (isTopReaderLoading || topReader.length === 0) {
+    return null;
+  }
+
+  return (
+    <ActivityContainer>
+      <ActivitySectionHeader
+        title="Badges"
+        subtitle="Understand more how this is computed in"
+        clickableTitle="daily.dev docs"
+        link={topReaderBadgeDocs}
+      />
+
+      {/* TODO: add a null state when the user has no badges */}
```

Comment:
We already have designs?

Reply:
Forgot to remove this TODO, it's just not going to be rendered
https://dailydotdev.slack.com/archives/C078JGTQ9AT/p1730506925874319